packages/x-scheduler/src/primitives/use-event-calendar/store.ts

```diff
@@ -86,30 +91,30 @@ export const selectors = {
     (events, visibleResources, adapter, { days, shouldOnlyRenderEventInOneCell }) => {
       const daysMap = new Map<
         string,
-        { events: CalendarEvent[]; allDayEvents: CalendarEventWithPosition[] }
+        { events: CalendarEventOccurence[]; allDayEvents: CalendarEventOccurenceWithPosition[] }
       >();
       for (const day of days) {
         const dayKey = adapter.format(day, 'keyboardDate');
         daysMap.set(dayKey, { events: [], allDayEvents: [] });
       }
-      // STEP 1: Sort events by start date
-      // We need to sort the events by start date to ensure they are processed in the correct order and the row indexes for the all day events are set in the correct order
-      const sortedEvents = events.slice().sort((a, b) => {
-        if (adapter.isBefore(a.start, b.start)) {
-          return -1;
-        }
-        if (adapter.isAfter(a.start, b.start)) {
-          return 1;
-        }
-        return 0;
-      });
-      // STEP 2: Skip events from resources that are not visible
-      for (const event of sortedEvents) {
+
+      // Collect ALL event instances (both recurring and non-recurring)
+      const instances: CalendarEventOccurence[] = [];
```

Comment:
Should it be something like `visibleOccurences`, same for the comment above.
To stick with "occurences" whenever we talk about the events with the recurring part expanded.